packages/plugins/i18n/admin/src/contentManagerHooks/utils/addLocaleToLinksSearch.js

```diff
@@ -6,6 +6,10 @@ const addLocaleToLinksSearch = (links, kind, contentTypeSchemas, locales, permis
   return links.map((link) => {
     const contentTypeUID = link.to.split(`/${kind}/`)[1];
 
+    const currentURL = new URL(window.location.href);
+
+    const currentLocale = currentURL.searchParams.get('plugins[i18n][locale]');
```

Comment:
This this can be calculated outside of the map function maybe? Also does it make sense to do it this way? Can we not pass it to the func?

Reply:
Totally agree of doing it outside of the map I'll do the necessary changes.

Reply:
For passing the locale to the function, I agree that it would be ideal to have the logic in one central location instead of duplicating it for both Single types and Content type. This approach would ensure consistency and maintainability by avoiding code duplication. So we can stay now with the current solution.